extra/mariabackup/xtrabackup.cc

```diff
@@ -5408,6 +5408,69 @@ static void rename_force(const char *from, const char *to) {
 }
 
 
+/**
+ * Given an ibd file path, this returns the space id.
+ * @param file The ibd file
+ * @return space id for the ibd file, -1 if an error.
+ */
+static
+ulint get_space_id(const char *file) {
+	// make sure this is an ibd file ...
+	if (!ends_with(file, ".ibd"))
+		die("Error: Requesting space id for a non-ibd file: %s", file);;
+
+	int fd = open(file, O_RDONLY, 0);
+	if (fd < 0)
+		die("Error: Unable to open file for reading: %s", file);
+
+	// read first page to the buffer ...
+	// we are reading the base backup ibd file, therefore,
+	// srv_page_size should work (no need to read page size
+	// from delta meta file)
+	unsigned char *page = new unsigned char[srv_page_size];
+	memset(page, 0, srv_page_size);
+
+	ulong bytes_read;
+	if ((bytes_read = read(fd, page, srv_page_size)) == srv_page_size) {
+		// read the space id ...
+		ulint space_id = mach_read_from_4(
+				page + FIL_PAGE_ARCH_LOG_NO_OR_SPACE_ID);
```

Comment:
The page checksum is not being validated here.

Starting with 86dc7b4d4cfe15a2d37f8b5f60c4fce5dba9491d it is more likely that data files will not contain valid tablespace identifiers.

I think that any fix of this needs to be carefully tested in such a way that incremental backups are being made during a DDL intensive workload.

Reply:
@dr-m Any ideas how to fix this?
@thejaka-kanewala Are you sure there can be no clash on `xtrabackup#<space_id>` after multiple incremental backups?